### PR TITLE
[query] Make TableKeyByAndAggregate bufferSize parameter required

### PR DESCRIPTION
**Pull request by chrisvittal**
Allow the default value to be controlled by a context flag.

### CODE REVIEW

**Comment by chrisvittal** [comment]
This is barely for power users. I hate it. And it should die as soon as it can.

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 1]
```diff
@@ -2498,7 +2498,7 @@ case class TableKeyByAndAggregate(
   expr: IR,
   newKey: IR,
   nPartitions: Option[Int] = None,
-  bufferSize: Int = 50) extends TableIR {
+  bufferSize: Int = -1) extends TableIR {
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala:2501]: just take out the default value?

**Review by tpoterba — CHANGES_REQUESTED** [review]
thanks!

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala (hunk 1 above)
Comment (chrisvittal) [reply hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: didn't know you could _do_ that in scala. Have default parameters and non-default parameters after them in a parameter list.

**Review by tpoterba — APPROVED** [review]

**Pull request merged**